core/src/main/java/org/elasticsearch/action/admin/indices/shrink/ShrinkRequest.java

```diff
@@ -138,4 +138,14 @@ public void source(BytesReference source) {
             throw new ElasticsearchParseException("failed to parse content type for shrink index source");
         }
     }
+
+    /**
+     * Sets the number of shard copies that must be started for each shard before the index creation
+     * operation returns.  Defaults to the default write consistency level specified by the
+     * {@link ReplicationOperation#WRITE_CONSISTENCY_LEVEL_SETTING} setting.
+     */
+    public void setWaitOnShards(String waitOnShards) {
+        this.shrinkIndexRequest.waitOnShards(Objects.requireNonNull(waitOnShards));
```

Comment:
I think if you are going to go to the trouble to call `requireNonNull` you should have an error message. Either we expect it to throw and NPE in the parser or stick a message on the exception.